### DISCUSSION TITLE
support exporting config asynchronously for cli

### DISCUSSION THREAD

**Discussion by manan** [mikro-orm/mikro-orm#4185] [General]
**Is your feature request related to a problem? Please describe.**
Cannot perform async operations to export CLI configuration (eg., want to fetch database password / connection details asynchronously from a secret store)

**Describe the solution you'd like**
Support exporting an async function that returns config options to use for the CLI

**Describe alternatives you've considered**
Considered hack-y ways to force async operations to be performed synchronously by throwing it in a different process. Don't like it.

**Additional context**
No other context, I think it'd be nice to support this. I can look into creating a PR for this.

**Comment by manan** [comment]
hmm, just dug through the code - looks like the use case is [already supported](https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/utils/ConfigurationLoader.ts#L38). should this be documented somewhere?

**Reply by B4nan** [reply]
It is documented

<img width="993" alt="image" src="https://user-images.githubusercontent.com/615580/229078324-e25435a0-d3df-41fa-810d-6dd25b40ca48.png">

https://mikro-orm.io/docs/installation

**Reply by B4nan** [reply]
I guess it could be a bit more verbose and we should mention the async callback variant too. PR welcome
